From 128904f606e253dac154c55a32851052b85b09a9 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Fri, 6 Nov 2015 10:47:15 +0800
Subject: [PATCH 0763/5242] MLK-11979 arm: imx: enable rpmsg on imx6sx
 platforms

commit  ec34ff4dba264403bdbb244649e53f0d8bcf8471 from
https://source.codeaurora.org/external/imx/linux-imx.git

enable rpmsg on imx6sx platforms
 - imx6sx sdb board
 - imx6sx sabreauto board

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                |    1 +
 arch/arm/boot/dts/imx6sx-sabreauto-m4.dts |   26 +++++++++++++++-----------
 arch/arm/boot/dts/imx6sx-sdb-m4.dts       |    4 ++++
 arch/arm/boot/dts/imx6sx.dtsi             |    5 +++++
 4 files changed, 25 insertions(+), 11 deletions(-)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 4a05a5f..bd5f111 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -552,6 +552,7 @@ dtb-$(CONFIG_SOC_IMX6SX) += \
 	imx6sx-sdb-btwifi.dtb \
 	imx6sx-sdb-lcdif1.dtb \
 	imx6sx-sdb-ldo.dtb \
+	imx6sx-sdb-m4.dtb \
 	imx6sx-sdb-mqs.dtb \
 	imx6sx-19x19-arm2.dtb \
 	imx6sx-19x19-arm2-ldo.dtb \
diff --git a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
index 159e907..51ac31c 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
@@ -20,28 +20,28 @@
  * The flollowing modules are conflicting with M4, disable them when m4
  * is running.
  */
-&flexcan1 {
-	status = "disabled";
+&adc1 {
+        status = "disabled";
 };
 
-&flexcan2 {
-	status = "disabled";
+&adc2 {
+        status = "disabled";
 };
 
-&i2c3 {
+&flexcan1 {
 	status = "disabled";
 };
 
-&uart2 {
+&flexcan2 {
 	status = "disabled";
 };
 
-&adc1 {
+&i2c3 {
 	status = "disabled";
 };
 
-&adc2 {
-	status = "disabled";
+&ocram {
+	reg = <0x00901000 0x1E000>;
 };
 
 &ocram {
@@ -57,8 +57,12 @@
 	status = "okay";
 };
 
-&ocram {
-	reg = <0x00901000 0x1E000>;
+&rpmsg{
+	status = "okay";
+};
+
+&uart2 {
+	status = "disabled";
 };
 
 &clks {
diff --git a/arch/arm/boot/dts/imx6sx-sdb-m4.dts b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
index d96544d..78f1b6b 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
@@ -55,6 +55,10 @@
 	status = "okay";
 };
 
+&rpmsg{
+	status = "okay";
+};
+
 &uart2 {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index f4a5885..9ee9bca 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1442,6 +1442,11 @@
 				status = "okay";
 			};
 
+			rpmsg: rpmsg{
+				compatible = "fsl,imx6sx-rpmsg";
+				status = "disabled";
+			};
+
 			uart6: serial@22a0000 {
 				compatible = "fsl,imx6sx-uart",
 					     "fsl,imx6q-uart", "fsl,imx21-uart";
-- 
1.7.9.5

